### PR TITLE
Introduce the RetryMessage inorder to re-queue the failed handled message

### PR DESCRIPTION
**Pull request by sagarrohankar-bsft**
Also introduce the commit interval as part of consumer creation.

### CODE REVIEW

File: pulsarlib/messaging.go [hunk 1]
```diff
@@ -12,6 +12,14 @@ import (
 	"github.com/apache/pulsar-client-go/pulsar"
 )
 
+// Error for retrying failed consumed message. Client can return this error to ensure the message
+// will be enqueued for a given RetryAfter duration.
+type HandleMessageError struct {
```
Comment (merajblueshift) [inline pulsarlib/messaging.go:17]: Name this RetryMessage

Comment (sagarrohankar-bsft) [reply pulsarlib/messaging.go]: Done

File: pulsarlib/messaging.go [hunk 2]
```diff
@@ -143,8 +151,16 @@ func (m *messaging) processMessageWorker() {
...
-		messageItem.pulsarc.Ack(messageItem.message)
+		retry := messageItem.handler.HandleMessage(m)
+		if retry != nil {
+			if retry.RetryAfter == 0 {
+				messageItem.pulsarc.Nack(messageItem.message)
+			} else {
+				messageItem.pulsarc.ReconsumeLater(messageItem.message, retry.RetryAfter)
+			}
+		} else {
+			messageItem.pulsarc.Ack(messageItem.message) // Error of diffrent type, so preserve the default of ack
```
Comment (merajblueshift) [inline pulsarlib/messaging.go:162]: Remove the comment. We don't know if it was an error or not. For us it is a success

Comment (sagarrohankar-bsft) [reply pulsarlib/messaging.go]: Done

**Pull request merged**